optimize/minimize.go

```diff
@@ -13,6 +13,18 @@ import (
 	"gonum.org/v1/gonum/mat"
 )
 
+var (
+	nonpositiveDimension string = "optimize: non-positive input dimension"
```

Comment:
I would call them `errNonPositiveDimension` and `errNegativeTasks`.
and create them with `errors.New(...)`

Reply:
These should be constants.